Add a note about the need of Basic Auth plugin to the docs.

tomalec opened the pull request:
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

Add a note about the need for the Basic Auth plugin to the docs.
`@woocommerce/e2e-utils.withRestApi` requires Basic Auth plugin to work, which was not explicitly stated in the docs before.

The only place where Basic Auth is mentioned is https://github.com/woocommerce/woocommerce/blob/trunk/tests/e2e/env/builtin.md#project-initialization.
But **that's a different package**, it's quite deep in its docs, seems to be just an example of "WooCommerce core's script", like an example of project-specific setup, and project's dependencies not `e2e-utils` API dependency, without an emphasis on the need for Basic Auth for any other project.

Related to https://github.com/woocommerce/woocommerce/issues/30687#issuecomment-919989929 and https://github.com/woocommerce/woocommerce/issues/30280

### How to test the changes in this Pull Request:

1. Read the `@woocommerce/e2e-utils` docs/README https://github.com/woocommerce/woocommerce/blob/update/e2e-utils-docs/tests/e2e/utils/README.md
2. Find the information about what's needed to make the `withRestApi` work.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [n/a] Have you written new tests for your changes, as applicable?
* [n/a] Have you successfully run tests with your changes locally?

<!-- Mark completed items with an [x] -->

### Changelog entry

> Mention the dependency on the Basic Auth plugin in the `withRestApi`'s docs.

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

zhongruige commented on tests/e2e/utils/README.md:
```diff
@@ -138,6 +138,10 @@ This package provides support for enabling retries in tests:
 
 ### REST API `withRestApi`
 
+To use this API you need to install [the Basic Auth plugin](https://github.com/WP-API/Basic-Auth),
```
Some suggestions for tweaking the wording a bit here:

> Please note: if you're using a non-SSL environment (such as a Docker container) you will need to use Basic Auth in order to authenticate with the API and use the `withRestApi` methods listed below. To do so, you will need to install the [the Basic Auth plugin](https://github.com/WP-API/Basic-Auth). One way this can be accomplished is by adding `wp plugin install https://github.com/WP-API/Basic-Auth/archive/master.zip --activate` to your `initialize.sh` script.

Let me know what you think!

zhongruige reviewed (commented):
Thanks for working on this @tomalec! I left a small comment on a wording change, let me know what you think.

tomalec commented:
Thanks, @zhongruige  and sorry, that it took me that long. I used your suggestion with a small addition of:
```
(such as a Docker container from [`wc-e2e`](https://www.npmjs.com/package/@woocommerce/e2e-environment))
```

As a developer just trying to set up `@woo…/e2e` suite, I'm not yet very familiar with the fact that it uses docker internally. So I may think: "Well… I didn't set up Docker myself, I just installed `woocommerce/e2e-environment` by the book, as [its README](https://www.npmjs.com/package/@woocommerce/e2e-environment) does not even mention Docker, so probably, that exception note is not for me."

If the note mentions `wc-e2e` not just docker, it got a higher chance to catch my attention.

tomalec committed:
Add a note about the need of Basic Auth plugin to the docs.

Related to https://github.com/woocommerce/woocommerce/issues/30687#issuecomment-919989929

tomalec committed:
Elaborate why Basic Auth is needed to use `e2e-utils.withRestApi`

Co-authored-by: Greg <71906536+zhongruige@users.noreply.github.com>

Addresses https://github.com/woocommerce/woocommerce/pull/30724#discussion_r715113390

tomalec commented:
I rebased and force pushed the branch, to get rid of conflicts, and make the history easier to grasp, after the monorepofication/ moving the PR from repo to repo.

zhongruige commented:
> I rebased and force pushed the branch, to get rid of conflicts, and make the history easier to grasp, after the monorepofication.

Thanks so much for taking care of this!

zhongruige commented:
> Thanks, @zhongruige and sorry, that it took me that long. I used your suggestion with a small addition of:
> 
> ```
> (such as a Docker container from [`wc-e2e`](https://www.npmjs.com/package/@woocommerce/e2e-environment))
> ```
> 
> As a developer just trying to set up `@woo…/e2e` suite, I'm not yet very familiar with the fact that it uses docker internally. So I may think: "Well… I didn't set up Docker myself, I just installed `woocommerce/e2e-environment` by the book, as [its README](https://www.npmjs.com/package/@woocommerce/e2e-environment) does not even mention Docker, so probably, that exception note is not for me."
> 
> If the note mentions `wc-e2e` not just docker, it got a higher chance to catch my attention.

Makes sense! I like this addition 👍

zhongruige reviewed (approved):
Changes look great, thanks for improving our documentation here @tomalec!
